docs/manual/developer/04_style_guide.md

```diff
@@ -15,7 +15,36 @@
 * Should use the [merge commit method](https://docs.github.com/en/github/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges)
 * Should use the GitHub Web UI to document and ensure code reviews are done correctly
 
-## All files
+
+## General Coding Style
+
+Prioritize the human-readability of code to its machine efficiency.
+In practice, this implies:
+
+* Keep functions and methods small, just a few of lines, so it is possible to understand what they do with minimal effort.
+  Prioritize the code readability over line count or code performance.
+* Don't mix low-level and high-level code in one function.
+  Extract functionality to separate functions or convert functions into classes with multiple methods to avoid this.
+* If a function contains try/except block, there should be no other logic present apart from exception handling.
+* Name variables and functions properly, don't hesitate to use longer identifiers.
+  Use explanatory variables to help the code to express itself -
+  assign a result of the operation into a variable that describes its meaning even if you intend to use it only once.
+* Don't copy-paste code, use e.g. jinja macros to reduce duplication.
+  Exception to this rule is code that is identical another piece of code only by coincidence, and there is a substantial probability that the code can diverge.
+* Include tests for your contribution.
+* Write comments if and only if there is no chance for the code to explain itself.
+* Don't put authorship information into the code, Git tracks authorship for you.
+  Don't copy-paste license text into source files -- use [SPDX IDs](https://spdx.dev/ids/) for that purpose.
+* Don't take part in making files longer - files longer than 400 lines should be an exception.
```

Comment:
```suggestion
* Don't take part in making files longer - files longer than 250 lines should be an exception.
```
Currently, Code Climate is set to a 250 LOC limit, we can change that to 400 if you think that is a better limit.

Reply:
So I would converge gradually, so let's set the limit to 400, and then restrict it further when there is a lower amount of findings.

Reply:
Sounds good; I will get a PR up with the code climate config.

Reply:
See #8785